modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/AbstractPlannerTest.java

```diff
@@ -1318,4 +1344,48 @@ public Publisher<BinaryRow> lookup(int partId, HybridTimestamp timestamp, Cluste
     Predicate<SearchBounds> empty() {
         return Objects::isNull;
     }
+
+    /**
+     * Wrapper for RelOptListener with empty realization for each of their methods. Good choice in case you need implement just one or few
+     * methods for listener.
+     */
+    static class RelOptListenerWrapper implements RelOptListener {
+
+        @Override
+        public void relEquivalenceFound(RelEquivalenceEvent event) {}
+
+        @Override
+        public void ruleAttempted(RuleAttemptedEvent event) {}
+
+        @Override
+        public void ruleProductionSucceeded(RuleProductionEvent event) {}
+
+        @Override
+        public void relDiscarded(RelDiscardedEvent event) {}
+
+        @Override
+        public void relChosen(RelChosenEvent event) {}
+    }
+
+    /**
+     * Listener which keeps information about attempts of applying optimization rules.
+     */
+    public class RuleAttemptListener extends RelOptListenerWrapper {
+        Set<String> attemptedRules = new HashSet<>();
+
+        @Override
+        public void ruleAttempted(final RuleAttemptedEvent event) {
+            if (event.isBefore()) {
+                attemptedRules.add(event.getRuleCall().getRule().toString());
+            }
+        }
+
+        public boolean isApplied(String rule) {
+            return attemptedRules.stream().anyMatch(r -> r.contains(rule));
+        }
+
+        public void resetStatistics() {
```

Comment:
statistic - confusing, probably simple - **reset** is enough ?

Reply:
ok